src/smefit/rge.py

```diff
@@ -0,0 +1,424 @@
+                )
+                _logger.warning(
+                    "Assuming it is a UV coupling and associating it to the null vector."
+                )
+                wc_basis[wc_name] = {}
+                continue
+
+            wc_warsaw_name = wcxf_dict["wc"]
+            if "value" not in wcxf_dict:
+                wc_warsaw_value = [1] * len(wcxf_dict["wc"])
+            else:
+                # check if value is gs
+                # (this is a special case for OtG)
+                if wcxf_dict["value"] == ["gs"]:
+                    wc_warsaw_value = [evolve_gs(self.init_scale)]
+                else:
+                    wc_warsaw_value = wcxf_dict["value"]
+
+            # 1e-6 is because the Warsaw basis is in GeV^-2
+            wc_value = {
+                wc: val * 1e-6 for wc, val in zip(wc_warsaw_name, wc_warsaw_value)
+            }
+            wc_basis[wc_name] = wc_value
+
+        return wc_basis
+
+    def map_to_smefit(self, wc_final_vals, scale):
+        """
+        Map the Wilson coefficients from the Warsaw basis to the SMEFiT basis.
+        """
+        wc_dict = {}
+        for wc_basis, wc_inv_dict in inverse_wcxf_translate.items():
+            wc_warsaw_name = wc_inv_dict["wc"]
+            if "coeff" not in wc_inv_dict:
+                wc_warsaw_coeff = [1] * len(wc_warsaw_name)
+            else:
+                # check if coeff is 1/gs
+                # (this is a special case for OtG)
+                if wc_inv_dict["coeff"] == ["1/gs"]:
+                    wc_warsaw_coeff = [1 / evolve_gs(scale)]
+                else:
+                    wc_warsaw_coeff = wc_inv_dict["coeff"]
+
+            value = 0.0
+            for wc, coeff in zip(wc_warsaw_name, wc_warsaw_coeff):
+                if wc in wc_final_vals:
+                    # 1e6 is to transform from GeV^-2 to TeV^2
+                    value += 1e6 * wc_final_vals[wc].real * coeff
+            wc_dict[wc_basis] = value
+        return wc_dict
+
+    @property
+    def all_ops(self):
+        return sorted(wcxf_translate.keys())
+
+    def RGEevolve(self, wcs, scale):
+        """
+        Evolve the Wilson coefficients from the initial scale to the scale of interest.
+        """
+        wc_wilson = {}
+        for op, values in self.RGEbasis.items():
+            for key in values:
+                if key not in wc_wilson:
+                    wc_wilson[key] = values[key] * wcs[op]
+                else:
+                    wc_wilson[key] += values[key] * wcs[op]
+
+        wc_init = wilson.Wilson(
+            wc_wilson, scale=self.init_scale, eft="SMEFT", basis="Warsaw"
+        )
+        wc_init.set_option("smeft_accuracy", self.accuracy)
+        wc_final = wc_init.match_run(scale=scale, eft="SMEFT", basis="Warsaw").dict
+
+        # remove small values
+        wc_final = {key: value for key, value in wc_final.items() if abs(value) > 1e-10}
+
+        return self.map_to_smefit(wc_final, scale)
```

Comment:
This code is not used, it got replaced by RGEmatrix_dict I think

Reply:
True, I think I wanted it because it's the equivalent of the function of wilson... Basically, it's not computing the RGE matrix, it's doing the actual evolution of a specific parameter space point.